### ISSUE TITLE
Development Environment Issues

### ISSUE THREAD

**Issue by aaron-pham** [sw-tools/checkin-service#11]
Hey,

Since I'd like to get everything nice and cleaned up as I'm working on this project, I want my development environment to be as "error" free as possible.

When running `npm run compile`, I get some TS errors:
```
npm run compile

> @sw-tools/check-in-service@1.0.0 compile /Users/aaronpham/repos/checkin-service
> rm -rf dist && tsc

src/lib/sw-generate-headers.ts:27:30 - error TS7006: Parameter 'request' implicitly has an 'any' type.

27   page.on('requestfinished', request => {
                                ~~~~~~~
src/lib/sw-generate-headers.ts:51:22 - error TS7006: Parameter 'request' implicitly has an 'any' type.

51   page.on('request', request => {
                        ~~~~~~~
Found 2 errors.
```

Also, in VSCode, .eslintrc.js complains with this error:
```
Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: .eslintrc.js.
The file must be included in at least one of the projects provided.
```

I fixed the above error by adding `.eslintrc.js` into my tsconfig.json:
```
{
  "extends": "@tsconfig/node16/tsconfig.json",
  "compileOnSave": true,
  "compilerOptions": {
    "lib": ["es2021", "dom"],
    "esModuleInterop": true,
    "checkJs": false,
    "allowJs": false,
    "strictNullChecks": false,
    "noImplicitAny": true,
    "useUnknownInCatchVariables": false,
    "outDir": "dist",
    "removeComments": true
  },
  "include": [
    "src",
    ".eslintrc.js"
  ]
}
```

Are you seeing any of these issues as well? Or is it related to my environment?

**Comment by swtools0** [comment]
I've added a .eslintignore file to avoid the eslint parsing error.

I'm not sure why TypeScript cannot infer the type of `request`. I don't see this problem in my environment. Make sure your npm libraries are updated and that the tsc binary you're running is `node_modules/.bin/tsc` and not a globally installed binary from a potentially different version of TypeScript.

**Comment by aaron-pham** [comment]
Ah yeah, the tsc binary was a different version. However, whenever I do npm install in the root directory, I noticed I got errors about peer dependencies that are required. 

Should these be added to the `package.json`? After doing `npm install puppeteer-core@^10.1.0`, my TS errors went away, however it was added to my `package.json`. 

Running `npm install` again, I got `npm WARN inquirer-autocomplete-prompt@1.4.0 requires a peer of inquirer@^5.0.0 || ^6.0.0 || ^7.0.0 || ^8.0.0 but none is installed. You must install peer dependencies yourself.`

This was also added to my `package.json`, and now when I run npm install, everything works fine. Not gunna lie, I am really confused with the npm package management (I come from the java/python world!). If this is missing from the dependencies, why would the code work on AWS?

Thanks for your replies!
